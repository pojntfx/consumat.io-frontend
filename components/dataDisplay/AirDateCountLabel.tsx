import { CalendarIcon } from "@heroicons/react/outline";
import React from "react";
import { Episode } from "../../lib/api/consumat-io";
import {
  daysUntilDate,
  isDateInFuture,
  isDateInPast,
  isDateInPresent,
} from "../../types/date";

type AirDateCountLabelProps = {
  episode: Episode;
  className?: string;
};

function AirDateCountLabel({ episode, className }: AirDateCountLabelProps) {
  return (
    <div className={"flex flex-row " + className}>
      {episode.airDate == null || episode.airDate == "" ? (
        <>
          <CalendarIcon className="h-6 w-5 ml-2 mr-1 text-gray-500" />
          <div className="font-medium truncate italic text-gray-500">
            coming
          </div>
        </>
      ) : isDateInPast(new Date(episode.airDate)) ? (
        <>
          <CalendarIcon className="h-6 w-5 ml-2 mr-1 text-gray-500" />
          <div className="font-medium truncate italic text-gray-500">
            released
          </div>
        </>
      ) : isDateInPresent(new Date(episode.airDate)) ? (
        <>
          <CalendarIcon className="h-6 w-5 ml-2 mr-1 text-gray-500" />
          <div className="font-medium truncate italic text-gray-500">today</div>
        </>
      ) : (
        isDateInFuture(new Date(episode.airDate)) && (
          <>
            <CalendarIcon className="h-6 w-5 ml-2 mr-1 text-gray-500" />
            <div className="font-medium truncate italic text-gray-500">
              {daysUntilDate(new Date(episode.airDate)) + " days"}
            </div>
          </>
        )
      )}
    </div>
  );
}

export default AirDateCountLabel;
