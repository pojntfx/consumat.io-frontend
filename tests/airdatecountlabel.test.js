import React from "react";
import renderer from "react-test-renderer";
import AirDateCountLabel from "../components/dataDisplay/AirDateCountLabel";
import testExamples from "./testExamples";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <AirDateCountLabel
        className={testExamples.exampleClassname}
        episode={testExamples.exampleEpisode}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
