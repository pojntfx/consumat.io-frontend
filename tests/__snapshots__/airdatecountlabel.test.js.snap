// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-row ml-2"
>
  <svg
    className="h-6 w-5 mr-1 text-gray-500"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth={2}
    />
  </svg>
  <div
    className="font-medium truncate italic text-gray-500"
  >
    released
  </div>
</div>
`;
